src/core/input.rs

```diff
@@ -17,6 +17,17 @@ pub enum MouseButton {
     X2,
 }
 
+fn map_mouse_button(button: winit::event::MouseButton) -> Option<MouseButton> {
+    match button {
+        winit::event::MouseButton::Left => Some(MouseButton::Left),
+        winit::event::MouseButton::Right => Some(MouseButton::Right),
+        winit::event::MouseButton::Middle => Some(MouseButton::Middle),
+        winit::event::MouseButton::Other(8) => Some(MouseButton::X1),
```

Comment:
I'm not familiar with mouse buttons beyond left/right/middle. What does 8 and 9 represent here? Is this hardware specific?

Reply:
Glad to help.
You had button X1 and X2 so I tried to filled them with the next two IDs. Tested on two devices and got 8 and 9 as the next button IDs after Left, Right and Middle. I don't know to what extent this is hardware specific. We could remove X1/X2 to avoid guessing.

Reply:
SDL: https://github.com/spurious/SDL-mirror/blob/release-2.0.12/include/SDL_mouse.h#L282
```
#define SDL_BUTTON_X1       4
#define SDL_BUTTON_X2       5
```
Which seems reasonable. But have also found other combinations 5,6 and 7,8

Reply:
Let's keep them for now, if people run into issues we can always fix it later. I'm going to merge this! 👍